packages/editor-ui/src/views/WorkflowsView.vue

```diff
@@ -26,303 +55,248 @@ interface Filters {
-		},
-		statusFilterOptions(): Array<{ label: string; value: string | boolean }> {
-			return [
-				{
-					label: this.$locale.baseText('workflows.filters.status.all'),
-					value: StatusFilter.ALL,
-				},
-				{
-					label: this.$locale.baseText('workflows.filters.status.active'),
-					value: StatusFilter.ACTIVE,
-				},
-				{
-					label: this.$locale.baseText('workflows.filters.status.deactivated'),
-					value: StatusFilter.DEACTIVATED,
-				},
-			];
-		},
-		userRole() {
-			const role = this.usersStore.currentUserCloudInfo?.role;
-
-			if (role) {
-				return role;
-			}
-
-			const answers = this.usersStore.currentUser?.personalizationAnswers;
-			if (answers && 'role' in answers) {
-				return answers.role;
-			}
-
-			return undefined;
-		},
-		isOnboardingExperimentEnabled() {
-			return (
-				this.posthogStore.getVariant(MORE_ONBOARDING_OPTIONS_EXPERIMENT.name) ===
-				MORE_ONBOARDING_OPTIONS_EXPERIMENT.variant
-			);
-		},
-		isSalesUser() {
-			if (!this.userRole) {
-				return false;
-			}
-			return ['Sales', 'sales-and-marketing'].includes(this.userRole);
-		},
-		addWorkflowButtonText() {
-			return this.projectsStore.currentProject
-				? this.$locale.baseText('workflows.project.add')
-				: this.$locale.baseText('workflows.add');
-		},
-		projectPermissions() {
-			return getResourcePermissions(
-				this.projectsStore.currentProject?.scopes ?? this.projectsStore.personalProject?.scopes,
-			);
-		},
-		emptyListDescription() {
-			if (this.readOnlyEnv) {
-				return this.$locale.baseText('workflows.empty.description.readOnlyEnv');
-			} else if (!this.projectPermissions.workflow.create) {
-				return this.$locale.baseText('workflows.empty.description.noPermission');
-			} else {
-				return this.$locale.baseText('workflows.empty.description');
-			}
-		},
+	{
+		label: i18n.baseText('workflows.filters.status.active'),
+		value: StatusFilter.ACTIVE,
 	},
-	watch: {
-		filters: {
-			deep: true,
-			handler() {
-				this.saveFiltersOnQueryString();
-			},
-		},
-		'$route.params.projectId'() {
-			void this.initialize();
-		},
+	{
+		label: i18n.baseText('workflows.filters.status.deactivated'),
+		value: StatusFilter.DEACTIVATED,
 	},
-	async mounted() {
-		this.documentTitle.set(this.$locale.baseText('workflows.heading'));
+]);
 
-		await this.tagsStore.fetchAll();
+const userRole = computed(() => {
+	const role = usersStore.currentUserCloudInfo?.role;
+	if (role) return role;
 
-		await this.setFiltersFromQueryString();
+	const answers = usersStore.currentUser?.personalizationAnswers;
+	if (answers && 'role' in answers) {
+		return answers.role;
+	}
 
-		void this.usersStore.showPersonalizationSurvey();
+	return undefined;
+});
 
-		this.sourceControlStoreUnsubscribe = this.sourceControlStore.$onAction(({ name, after }) => {
-			if (name === 'pullWorkfolder' && after) {
-				after(() => {
-					void this.initialize();
-				});
-			}
-		});
-	},
-	beforeUnmount() {
-		this.sourceControlStoreUnsubscribe();
```

Comment:
this is not needed since pinia will remove the listener on it's own once the component is destroyed, unless explicitly told not to https://pinia.vuejs.org/core-concepts/actions.html#Subscribing-to-actions